src/lib.rs

```diff
@@ -409,6 +417,7 @@ fn left_len(content_len: usize) -> usize {
 // right side off to another thread, if idle threads are available. If the
 // "rayon" feature is disabled, just make ordinary function calls for the left
 // and the right.
+#[inline]
```

Comment:
Out of curiosity, are templated functions always effectively inline?

Reply:
> Out of curiosity, are templated functions always effectively inline?

No, they're only monomorphized. However, both gcc and llvm love to inline functions which are only called in a single place, and this function is only called in a single place. Even if later it's called in more than one place, it might still be inlined if LLVM decides it's too small. The only place where `#[inline]` is really mandatory is for cross-crate inlining (at least for now, perhaps in the future rustc might detect small functions at the MIR level and automatically make them inline).